### PR TITLE
fix: updated baseHRefRegex to perform lazy match

### LINKED ISSUE #9660: ArgoCD UI: Wrong index.html when running on subpath
* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

The index.html's content is malformed when running the server with --basehref and --rootpath different from default ("/"). 

When I run the server using "argocd" as value to --basehref and --rootpath I get the following result:
```html
<!doctype html><html lang="en"><head><meta charset="UTF-8"><title>Argo CD</title><base href="/argocd/"></div></body></html>
```

This is happening because the function `getIndexData` is  replacing more than just the base href.

https://github.com/argoproj/argo-cd/blob/91aefabc5b213a258ddcfe04b8e69bb4a2dd2566/server/server.go#L886-L901

Since the ArgoCD UI is now been built using the production mode and thus the html's white space are removed, the `baseHRefRegex` should be a non-greedy regex, so using 

```go
baseHRefRegex    = regexp.MustCompile(`<base href="(.*?)">`)
```  
instead of

```go
baseHRefRegex    = regexp.MustCompile(`<base href="(.*)">`)
``` 
will probably fix the bug (no tested yet)

**To Reproduce**

Just run the server using "argocd" as  --basehref and --rootpath

**Expected behavior**

Only the base href should be replaced in the index.html

**Screenshots**

**Version**

```shell
argocd@argocd-server-5674bcbc44-f5qwx:~$ argocd version
argocd: v2.4.0+91aefab
  BuildDate: 2022-06-10T17:23:37Z
  GitCommit: 91aefabc5b213a258ddcfe04b8e69bb4a2dd2566
  GitTreeState: clean
  GoVersion: go1.18.3
  Compiler: gc
  Platform: linux/amd64
```

**Logs**

```
Paste any relevant application logs here.
```

### LINKED ISSUE #9692: ArgoCD 2.4.0 webUI with rootpath no longer works.
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

Existing 2.3.4 installation upgraded to 2.40 using a `rootpath` of "/argocd" now  Web UI no longer renders.  ArgoCD itself seems to working with the repo just fine and still processing changes.  Just the Web UI is not rendering.

I also upgraded the CRDS:

```shell
$ kubectl apply -k https://github.com/argoproj/argo-cd.git/manifests/crds?ref=v2.4.0

customresourcedefinition.apiextensions.k8s.io/applications.argoproj.io unchanged
customresourcedefinition.apiextensions.k8s.io/applicationsets.argoproj.io configured
customresourcedefinition.apiextensions.k8s.io/appprojects.argoproj.io unchanged
```

**To Reproduce**

The 2.3.4 Helm `values.yaml` had this:

```yaml
  server:
    name: server
    extraArgs:
      - --insecure
      - --rootpath=/argocd
```

In reviewing logs after 2.4.0 upgrade I noticed this conflict warning:

```shell
$ k logs argocd-server-66497bbc98-j8f7s -n argocd -f
time="2022-06-16T12:43:20Z" level=warning msg="--basehref and --rootpath had conflict: basehref: / rootpath: /argocd"
time="2022-06-16T12:43:20Z" level=info msg="Starting configmap/secret informers"
time="2022-06-16T12:43:21Z" level=info msg="Configmap/secret informer synced"
time="2022-06-16T12:43:21Z" level=info msg="Starting configmap/secret informers"
time="2022-06-16T12:43:21Z" level=info msg="secrets informer cancelled"
time="2022-06-16T12:43:21Z" level=info msg="Configmap/secret informer synced"
time="2022-06-16T12:43:21Z" level=info msg="configmap informer cancelled"
time="2022-06-16T12:43:21Z" level=info msg="argocd v2.4.0+91aefab serving on port 8080 (url: , tls: false, namespace: argocd, sso: false)"
time="2022-06-16T12:43:21Z" level=info msg="0xc000f7de00 subscribed to settings updates"
time="2022-06-16T12:43:21Z" level=info msg="Starting rbac config informer"
time="2022-06-16T12:43:21Z" level=info msg="RBAC ConfigMap 'argocd-rbac-cm' added"
```

Educated guess, I set that as well:

```yaml
  server:
    name: server
    extraArgs:
      - --insecure
      - --rootpath=/argocd
      - --basehref=/argocd
```

Conflict warning message went away, but still Web UI does not render.
```shell
$ k logs argocd-server-b47dfccd-jxp8q -n argocd -f
time="2022-06-16T13:31:56Z" level=info msg="Starting configmap/secret informers"
time="2022-06-16T13:31:57Z" level=info msg="Configmap/secret informer synced"
time="2022-06-16T13:31:57Z" level=info msg="Starting configmap/secret informers"
time="2022-06-16T13:31:57Z" level=info msg="secrets informer cancelled"
time="2022-06-16T13:31:57Z" level=info msg="configmap informer cancelled"
time="2022-06-16T13:31:57Z" level=info msg="Configmap/secret informer synced"
time="2022-06-16T13:31:57Z" level=info msg="argocd v2.4.0+91aefab serving on port 8080 (url: , tls: false, namespace: argocd, sso: false)"
time="2022-06-16T13:31:57Z" level=info msg="0xc000f8af60 subscribed to settings updates"
time="2022-06-16T13:31:57Z" level=info msg="Starting rbac config informer"
time="2022-06-16T13:31:57Z" level=info msg="RBAC ConfigMap 'argocd-rbac-cm' added"
```

When I view source code of blank page, it is just this:
```html
<!doctype html><html lang="en"><head><meta charset="UTF-8"><title>Argo CD</title><base href="/argocd/"></div></body></html>
```

I can't find any errors to troubleshoot.

### PR DESCRIPTION
**Pull request by gdsoumya**
Signed-off-by: Soumya Ghosh Dastidar <gdsoumya@gmail.com>

Fixes #9660 
Fixes #9692 

This PR fixes the baseHRefRegex to perform a lazy match preventing index.html from being broken

After change resulting index.html (--rootpath=/argocd):
```
<!doctype html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Argo CD</title>
    <base href="/argocd/">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <link rel="icon" type="image/png" href="assets/favicon/favicon-32x32.png" sizes="32x32" />
    <link rel="icon" type="image/png" href="assets/favicon/favicon-16x16.png" sizes="16x16" />
    <link href="assets/fonts.css" rel="stylesheet">
    <script defer="defer" src="main.ee4eaaa358b34122032e.js"></script>
</head>
<body>
    <noscript>
        <p>Your browser does not support JavaScript. Please enable JavaScript to view the site. Alternatively, Argo CD
            can be used with the <a href="https://argoproj.github.io/argo-cd/cli_installation/">Argo CD CLI</a>.</p>
    </noscript>
    <div id="app"></div>
</body>
</html>
```

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -132,7 +132,7 @@ var backoff = wait.Backoff{
 
 var (
 	clientConstraint = fmt.Sprintf(">= %s", common.MinClientVersion)
-	baseHRefRegex    = regexp.MustCompile(`<base href="(.*)">`)
+	baseHRefRegex    = regexp.MustCompile(`<base href="(.*?)">`)
```
Comment (alexmt) [inline server/server.go:135]: Can you please add a test as well? I would suggest moving the line below into function and add a unit test for it 

https://github.com/argoproj/argo-cd/blob/a67b97d6ad5d17137190f6d9b863757da97e7c82/server/server.go#L896

**Review by alexmt — CHANGES_REQUESTED** [review]

**Comment by alexmt** [comment]
I think this is a regression, so adding cherry-pick label cc @crenshaw-dev , @leoluz

File: server/server.go (hunk 1 above)
Comment (gdsoumya) [reply server/server.go:135]: added unit test

**Review by alexmt — APPROVED** [review]
Thank you!

**Pull request merged**